components/core/src/clp/CurlGlobalInstance.cpp

```diff
@@ -0,0 +1,47 @@
+#include "CurlGlobalInstance.hpp"
+
+#include <cstddef>
+#include <mutex>
+
+#include <curl/curl.h>
+
+#include "CurlOperationFailed.hpp"
+#include "ErrorCode.hpp"
+
+namespace clp {
+std::mutex CurlGlobalInstance::m_global_mutex;
```

Comment:
It seems we can initialize the variable in the header file or using inline keyword.  A reference post is here

https://stackoverflow.com/questions/5019856/initialize-static-variables-in-c-class

Reply:
@kirkrodrigues Do u prefer inline or in the sec file?

Reply:
Using inline for now